napari/layers/_pyramid_layer/model.py

```diff
@@ -0,0 +1,249 @@
+
+        pos_in_slice = (self.coordinates[-2:] +
+                        self.translate[[1, 0]] / self.scale[:2])
+
+        # Make sure pos in slice doesn't go off edge
+        shape = self._image_shapes[self._pyramid_level]
+        coord = np.clip(pos_in_slice, 0, np.asarray(shape) - 1)
+        coord = np.round(coord*self.scale[:2]).astype(int)
+
+        return coord, value
+
+    def compute_pyramid_level(self, camera):
+        """Computed what level of the pyramid should be viewed given the
+        current camera position.
+
+        Parameters
+        ----------
+        camera : Camera
+            Vispy PanZoomCamera object.
+
+        Returns
+        ----------
+        level : int
+            Level of the pyramid to be viewing.
+        """
+        # Requested field of view from the camera in log units
+        size = np.log2(np.array(camera.rect.size).max())
+
+        # Max allowed tile in log units
+        max_size = np.log2(self._max_tile_shape.max())
+
+        # Allow for 2x coverage of field of view with max tile
+        diff = size - max_size + 1
+
+        # Find closed downsample level to diff
+        level = np.argmin(abs(np.log2(self._image_downsamples)-diff))
+
+        return level
+
+    def find_top_left(self):
+        """Finds the top left pixel of the canvas. Depends on the current
+        pan and zoom position
+
+        Returns
+        ----------
+        top_left : np.ndarry
+            Length two array with coordinates of top left pixel.
+        """
+
+        # Find image coordinate of top left canvas pixel
+        transform = self._node.canvas.scene.node_transform(self._node)
+        pos = transform.map([0, 0])[:2] + self.translate[:2]/self.scale[:2]
+
+        shape = self._image_shapes[0]
+
+        # Clip according to the max image shape
+        pos = [np.clip(pos[1], 0, shape[0]-1),
+               np.clip(pos[0], 0, shape[1]-1)]
+
+        # Convert to offset for image array
+        top_left = np.array(pos)
+        scale = self._max_tile_shape/4
+        top_left = scale*np.floor(top_left/scale)
+
+        return top_left.astype(int)
+
+    def get_message(self, coord, value):
+        """Generate a status message based on the coordinates and information
+        about what shapes are hovered over
+
+        Parameters
+        ----------
+        coord : sequence of int
+            Position of mouse cursor in image coordinates.
+        value : int or float or sequence of int or float
+            Value of the data at the coord.
+
+        Returns
+        ----------
+        msg : string
+            String containing a message that can be used as a status update.
+        """
+
+        msg = f'{coord}, {self.pyramid_level}, {self.name}' + ', value '
+        if isinstance(value, np.ndarray):
+            if isinstance(value[0], np.integer):
+                msg = msg + str(value)
+            else:
+                v_str = '[' + str.join(', ', [f'{v:0.3}' for v in value]) + ']'
```

Comment:
```suggestion
                v_str = '[' + ', '.join(f'{v:0.3}' for v in value) + ']'
```

Reply:
potentially better (have not investigated): `np.array_str(value, precision=3)` or `np.array_repr`

Reply:
I'd say this is for #93 and @jni to address